### ISSUE TITLE
IntelliSense and Snippet Stop working after few saves. 

### ISSUE THREAD

**Issue by aprather51** [Microsoft/vscode#35847]
- VSCode Version: Code 1.17.0
- OS Version: Windows_NT x64 10.0.15063

After recent update, the intelliSense and Snippet stop working on any languages after few saves. Turning off extensions does not solve problems.

**Comment by coreylendo** [comment]
i never signed up for this account I would appericate you finding out and getting back to me 850-999-3391 perior number 850-346-8875 also mail setting perior were coreylendo@yahoo.com i would alsi be greatful to have you help me find the intuder 

thanks

C.L

**Comment by coreylendo** [comment]
someone is using u to hack my accounts

C.L

**Comment by coreylendo** [comment]
please HELP me

C.L

**Comment by prashantham** [comment]
I have same issue. Saving takes a longer time as well. Noticed on 2 different macs.

**Comment by chrmarti** [comment]
Could you post the list of extensions you have installed? (`code --list-extensions`)

**Comment by fedemarco** [comment]
I have found that after I try import a file, it just stops working. I stop getting suggestions, and the intellisense stops working, along with the auto format, along with some plugins.
The import suggestions start working, and suddenly stop.

I removed all extensions and it happens anyway.

**Comment by prashantham** [comment]
code --list-extensions

CraigMaslowski.erb
abusaidm.html-snippets
dbaeumer.vscode-eslint
dhoeric.ansible-vault
donjayamanne.githistory
formulahendry.auto-close-tag
formulahendry.auto-rename-tag
haaaad.ansible
joshpeng.theme-onedark-sublime
mjmcloug.vscode-elixir
rebornix.Ruby
robertohuertasm.vscode-icons
xabikos.JavaScriptSnippets
xabikos.ReactSnippets
zhuangtongfa.Material-theme

**Comment by jrieken** [comment]
> After recent update, the intelliSense and Snippet stop working on any languages after few saves. 

What does that mean? Do you see loading forever and ever? Does it only suggest words and nothing smart? Do other language features, like error squiggles, still work?

**Comment by Jimbo80** [comment]
Same issue here. IntelliSense (object props, imports, user snippets) stops working after a while in 1.17. If i try to force IntelliSense by ctrl+space "loading..." is displayed but no result. 
It's also not possible to navigate to imports using F12, the blue loading indicator is displayed but nothing happens.

No extensions enabled and nothing in the developer tools console.

Squiggles still works. But light bulb for e.g. eslint fixes is not shown.

**Comment by jrieken** [comment]
@Jimbo80 Thanks for the info. It sounds like the language service, that is providing the smarts, stopped responding. Is this in JavaScript/TypeScript or another language?

**Comment by Jimbo80** [comment]
@jrieken It's JavaScript.
I might figured out how to reproduce this. Seems like it has to do with imports and incorrect paths.
If I try to write an import like: 
`import SearchInput from 'src/SearchInput';`

But misspell the path to something that doesn't exist and hit ctrl+space:
`import SearchInput from 'scr/<press ctrl+space for suggestions> ';` 

This seems to trigger the issue for me.

**Comment by jrieken** [comment]
Are any of VS Code child-process running busy? Can you craft a small sample that reproduces this?

**Comment by Jimbo80** [comment]
I don't see any increase in process activity for intellisens "loading...", but for F12 navigation to imports one of the child processes is running busy. 
I'm trying to create a small sample but I have not been able to reproduce the issue in the sample yet.

**Comment by azterix** [comment]
Seeing the same issue here. Was able to replicate the issue once using @Jimbo80's steps, but only the first time for some reason. Lock appeared to happen when I hit save. 

extensions: 
abusaidm.html-snippets
akamud.vscode-theme-onedark
alefragnani.project-manager
be5invis.vscode-custom-css
christian-kohler.npm-intellisense
christian-kohler.path-intellisense
CoenraadS.bracket-pair-colorizer
dbaeumer.vscode-eslint
donjayamanne.githistory
eg2.tslint
eg2.vscode-npm-script
esbenp.prettier-vscode
formulahendry.auto-close-tag
humao.rest-client
ilich8086.classic-asp
leizongmin.node-module-intellisense
lonefy.vscode-JS-CSS-HTML-formatter
mrmlnc.vscode-scss
ms-mssql.mssql
ms-vscode.atom-keybindings
ms-vscode.csharp
ms-vscode.PowerShell
msjsdiag.debugger-for-chrome
patrys.vscode-code-outline
PeterJausovec.vscode-docker
PKief.material-icon-theme
robinbentley.sass-indented
Shan.code-settings-sync
TwentyChung.jsx
wesbos.theme-cobalt2
xabikos.JavaScriptSnippets
zhuangtongfa.Material-theme

**Comment by patrick-mackay** [comment]
I'm experiencing the same behavior Jimbo80 described. I'm using typescript and was doing some refactoring and changing file paths on imports (and renames). 

Neither renaming (f2), find reference (f12) nor intellisense worked alter a while (a minute or so) and I found that one of the code.exe processes where using one cpu at something near 80%.

aurelia|Aur|0.3.4
Aurelia|beh|1.2.2
vscode-generate-getter-setter|DSK|0.4.2
gitlens|eam|5.6.0
tslint|eg2|1.0.11
Ionide-fsharp|Ion|3.5.1
Angular2|joh|2.7.0
azure-functions-tools|joh|0.3.1
vscode-JS-CSS-HTML-formatter|lon|0.2.3
vscode-apache|mrm|1.1.1
mssql|ms-|1.2.0
csharp|ms-|1.12.1
PowerShell|ms-|1.4.3
azurerm-vscode-tools|msa|0.3.7
standard|shi|0.2.3
nativescript|Tel|0.7.2
python|tht|0.2.3

**Comment by mjbvz** [comment]
@azterix / @patrick-mackay Do this still occur if you disable all extensions? If it does, can you please try getting the full command line arguments of the process that has the high cpu usage. This will help us track down what is causing this issue

**Comment by patrick-mackay** [comment]
@mjbvz , with extensions disabled the issue does not occur. All renamings, find reference and intellisense work as expected.

This is the command line with extensions enabled of the process.

"C:\Program Files\Microsoft VS Code\Code.exe" --type=renderer --js-flags=--nolazy --no-sandbox --primordial-pipe-token=FA4E83C44A93426C4A413550185752D9 --lang=en-US --app-user-model-id=Microsoft.VisualStudioCode --app-path="C:\Program Files\Microsoft VS Code\resources\app" --node-integration=true --webview-tag=true --no-sandbox --background-color=#1e1e1e --disable-blink-features=Auxclick --enable-pinch --device-scale-factor=1 --num-raster-threads=4 --enable-main-frame-before-activation --content-image-texture-target=0,0,3553;0,1,3553;0,2,3553;0,3,3553;0,4,3553;0,5,3553;0,6,3553;0,7,3553;0,8,3553;0,9,3553;0,10,3553;0,11,3553;0,12,3553;0,13,3553;0,14,3553;0,15,3553;1,0,3553;1,1,3553;1,2,3553;1,3,3553;1,4,3553;1,5,3553;1,6,3553;1,7,3553;1,8,3553;1,9,3553;1,10,3553;1,11,3553;1,12,3553;1,13,3553;1,14,3553;1,15,3553;2,0,3553;2,1,3553;2,2,3553;2,3,3553;2,4,3553;2,5,3553;2,6,3553;2,7,3553;2,8,3553;2,9,3553;2,10,3553;2,11,3553;2,12,3553;2,13,3553;2,14,3553;2,15,3553;3,0,3553;3,1,3553;3,2,3553;3,3,3553;3,4,3553;3,5,3553;3,6,3553;3,7,3553;3,8,3553;3,9,3553;3,10,3553;3,11,3553;3,12,3553;3,13,3553;3,14,3553;3,15,3553;4,0,3553;4,1,3553;4,2,3553;4,3,3553;4,4,3553;4,5,3553;4,6,3553;4,7,3553;4,8,3553;4,9,3553;4,10,3553;4,11,3553;4,12,3553;4,13,3553;4,14,3553;4,15,3553 --service-request-channel-token=FA4E83C44A93426C4A413550185752D9 --renderer-client-id=4 --mojo-platform-channel-handle=2152 /prefetch:1

**Comment by patrick-mackay** [comment]
After enabling all the extensions, 

I opened the Developer Tools and found that after renaming the file, adjunting the import accordingly and try a Goto Definition, if failed but this was logged on the console.

[msazurermtools.azurerm-vscode-tools] Cannot read property 'indexOf' of null: TypeError: Cannot read property 'indexOf' of null
	at Function.AppInsightsAsyncCorrelatedErrorWrapper.orig.prepareStackTrace (C:\Users\Patrick\.vscode\extensions\msazurermtools.azurerm-vscode-tools-0.3.7\node_modules\applicationinsights\out\AutoCollection\CorrelationContextManager.js:155:47)
	at Function.NativeError.prepareStackTrace (C:\Users\Patrick\.vscode\extensions\msazurermtools.azurerm-vscode-tools-0.3.7\node_modules\zone.js\dist\zone-node.js:900:30)
	at consoleError (C:\Users\Patrick\.vscode\extensions\msazurermtools.azurerm-vscode-tools-0.3.7\node_modules\zone.js\dist\zone-node.js:420:240)
	at _loop_1 (C:\Users\Patrick\.vscode\extensions\msazurermtools.azurerm-vscode-tools-0.3.7\node_modules\zone.js\dist\zone-node.js:449:25)
	at drainMicroTaskQueue (C:\Users\Patrick\.vscode\extensions\msazurermtools.azurerm-vscode-tools-0.3.7\node_modules\zone.js\dist\zone-node.js:453:21)
	at <anonymous>
e.onUnexpectedError @ shell.ts:454
(anonymous) @ shell.ts:414
e.onUnexpectedError @ errors.ts:119
o @ errors.ts:138
e.$onUnexpectedError @ mainThreadErrors.ts:26
e.invoke @ abstractThreadService.ts:41
e._invokeHandler @ rpcProtocol.ts:109
e._receiveOneMessage @ rpcProtocol.ts:96
(anonymous) @ rpcProtocol.ts:32
(anonymous) @ rpcProtocol.ts:157
e.invoke @ callbackList.ts:36
e.fire @ event.ts:123
(anonymous) @ ipc.net.ts:82
emitOne @ events.js:96
emit @ events.js:191
readableAddChunk @ _stream_readable.js:178
Readable.push @ _stream_readable.js:136
onread @ net.js:560

I disabled that extension, reloaded, and tried it again. Now it logged this on the console, but the renaming, gotos and intellisense worked as usual.

shell.ts:454 Cannot read property 'length' of undefined: TypeError: Cannot read property 'length' of undefined
	at DefinitionProvider.containsItem (C:\Users\Patrick\.vscode\extensions\DSKWRK.vscode-generate-getter-setter-0.4.2\out\src\provider.js:52:48)
	at C:\Users\Patrick\.vscode\extensions\DSKWRK.vscode-generate-getter-setter-0.4.2\out\src\bulb.js:10:56
	at Promise (<anonymous>)
	at CompleteActionProvider.provideCodeActions (C:\Users\Patrick\.vscode\extensions\DSKWRK.vscode-generate-getter-setter-0.4.2\out\src\bulb.js:7:16)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:28:372505
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:28:66174
	at new n.Class.derive._oncancel (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:28:51364)
	at Object.t.asWinJsPromise (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:28:66137)
	at e.provideCodeActions (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:28:372459)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:28:383686
e.onUnexpectedError @ shell.ts:454
(anonymous) @ shell.ts:414
..... (removed)

I disabled that extension, reloaded, and tried it again. Now it logged this on the console, but the renaming, gotos and intellisense worked as usual.

shell.ts:454 [eamodio.gitlens] this is not defined: ReferenceError: this is not defined
	at GitUri (C:\Users\Patrick\.vscode\extensions\eamodio.gitlens-5.6.0\out\src\git\gitUri.js:32:13)
	at GitContextTracker.<anonymous> (C:\Users\Patrick\.vscode\extensions\eamodio.gitlens-5.6.0\out\src\git\gitContextTracker.js:73:60)
	at Generator.next (<anonymous>)
	at C:\Users\Patrick\.vscode\extensions\eamodio.gitlens-5.6.0\out\src\git\gitContextTracker.js:7:71
	at Promise (<anonymous>)
	at __awaiter (C:\Users\Patrick\.vscode\extensions\eamodio.gitlens-5.6.0\out\src\git\gitContextTracker.js:3:12)
	at Timeout.setTimeout [as _onTimeout] (C:\Users\Patrick\.vscode\extensions\eamodio.gitlens-5.6.0\out\src\git\gitContextTracker.js:70:26)
	at ontimeout (timers.js:386:14)
	at tryOnTimeout (timers.js:250:5)
	at Timer.listOnTimeout (timers.js:214:5)
e.onUnexpectedError @ shell.ts:454
(anonymous) @ shell.ts:414
e.onUnexpectedError @ errors.ts:119
o @ errors.ts:138
e.$onUnexpectedError @ mainThreadErrors.ts:26
e.invoke @ abstractThreadService.ts:41
e._invokeHandler @ rpcProtocol.ts:109
e._receiveOneMessage @ rpcProtocol.ts:96
(anonymous) @ rpcProtocol.ts:32
(anonymous) @ rpcProtocol.ts:157
e.invoke @ callbackList.ts:36
e.fire @ event.ts:123
(anonymous) @ ipc.net.ts:82
emitOne @ events.js:96
emit @ events.js:191
readableAddChunk @ _stream_readable.js:178
Readable.push @ _stream_readable.js:136
onread @ net.js:560

After many attemps, I can say that having this extension enabled (msazurermtools.azurerm-vscode-tools) was hanging my installation. 

In the last hang, this are de processes that are using the CPU.
![image](https://user-images.githubusercontent.com/4219141/31469992-5cfab442-aeba-11e7-81b6-0ac9c03df132.png)

I guess that the gpu process is consuming 3% since it's displaying that horizontal line scrolling

![image](https://user-images.githubusercontent.com/4219141/31470027-8989b8c8-aeba-11e7-9f34-b6fe7883e414.png)

After switching windows, the CPU goes normal again... but the issue stills occurs.

![image](https://user-images.githubusercontent.com/4219141/31470076-bbd1b75e-aeba-11e7-9d4b-0955ee9e69e9.png)

"C:\Program Files\Microsoft VS Code\Code.exe" --type=renderer --js-flags=--nolazy --no-sandbox --primordial-pipe-token=CA2B0A01A28AAC351C7E721157CCE6B1 --lang=en-US --app-user-model-id=Microsoft.VisualStudioCode --app-path="C:\Program Files\Microsoft VS Code\resources\app" --node-integration=true --webview-tag=true --no-sandbox --background-color=#1e1e1e --disable-blink-features=Auxclick --enable-pinch --device-scale-factor=1 --num-raster-threads=4 --enable-main-frame-before-activation --content-image-texture-target=0,0,3553;0,1,3553;0,2,3553;0,3,3553;0,4,3553;0,5,3553;0,6,3553;0,7,3553;0,8,3553;0,9,3553;0,10,3553;0,11,3553;0,12,3553;0,13,3553;0,14,3553;0,15,3553;1,0,3553;1,1,3553;1,2,3553;1,3,3553;1,4,3553;1,5,3553;1,6,3553;1,7,3553;1,8,3553;1,9,3553;1,10,3553;1,11,3553;1,12,3553;1,13,3553;1,14,3553;1,15,3553;2,0,3553;2,1,3553;2,2,3553;2,3,3553;2,4,3553;2,5,3553;2,6,3553;2,7,3553;2,8,3553;2,9,3553;2,10,3553;2,11,3553;2,12,3553;2,13,3553;2,14,3553;2,15,3553;3,0,3553;3,1,3553;3,2,3553;3,3,3553;3,4,3553;3,5,3553;3,6,3553;3,7,3553;3,8,3553;3,9,3553;3,10,3553;3,11,3553;3,12,3553;3,13,3553;3,14,3553;3,15,3553;4,0,3553;4,1,3553;4,2,3553;4,3,3553;4,4,3553;4,5,3553;4,6,3553;4,7,3553;4,8,3553;4,9,3553;4,10,3553;4,11,3553;4,12,3553;4,13,3553;4,14,3553;4,15,3553 --service-request-channel-token=CA2B0A01A28AAC351C7E721157CCE6B1 --renderer-client-id=20 --mojo-platform-channel-handle=4588 /prefetch:1

**Comment by Jimbo80** [comment]
I finally been able to reproduce the issue in "small" sample, but only after including all dependencies from one of our large projects in package.json.
Further more we are only able to reproduce the issue when using yarn, using npm will not trigger this for us!? 😕 

Repo: https://github.com/Jimbo80/35847.git

Steps:
1. Use yarn to install dependencies.
2. Hammer the import in src/index.js. It's not super easy to trigger, you must save + type + save + ctrl+space + save.... and so on. Sometimes it's triggered immediately, sometimes you need to work for it :) Feels like the faster you type and save etc, the easier it is to reproduce.

We have been able to reproduce this on three different computers (win10 and server 2016) using this repo. Extensions does not seem to matter for us.

**Comment by azterix** [comment]
Not sure if it's related but I'm using yarn as well. I haven't tested without extensions yet, but I did notice that when triggered the condition just now and I mouse over what I just imported the ~tooltip popup is stuck on a 'Loading...' message:
<img width="491" alt="2017-10-12 16_44_47-clipboard" src="https://user-images.githubusercontent.com/1749872/31518371-07c09eac-af6d-11e7-8112-b88bf24b98d0.png">

This is also a project with a lot of dependencies pulled in directly under node_modules (946 items).

**Comment by bodiddlie** [comment]
Having the same issue with mine. It seems to occur when trying to run Intellisense on import statements in local relative paths. Working in an existing file without touching imports everything was working fine. I created a new .jsx file and did:

```
import React from 'react';
import moment from 'moment';
import styled from 'styled-components';

import {InputGroup, Label} from './field-wrap';
```
and the intellisense blew up. Trying to invoke it with ctrl+space just presents "Loading...". 

I've been able to duplicate this multiple times with these lines. Sometimes it blows up on the second line, sometimes not until the last one. Like @Jimbo80 said, it seems like the faster you type the import statements the easier it is to blow up. 

I'm running on Win10 using yarn for package management, and have gotten this to break with extensions enabled as well as disabled. I'm not seeing any error messages in the dev tools console.

**Comment by mjbvz** [comment]
Seems like multiple separate issues are being reported in this thread. Please try out [vscode insiders](https://code.visualstudio.com/insiders/) to see if your issue has been resolved. If it has not,  open a new issue so that we better track the individual problems being hit

---

@aprather51 Do you still see this bug on VS Code insiders?
